src/Microsoft.AspNetCore.CertificateGeneration.Task/CertificateGenerator.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.AspNetCore.CertificateGeneration.Task
+{
+    internal static class CertificateManager
+    {
+        public static X509Certificate2 GenerateSSLCertificate(
+            string subjectName,
+            IEnumerable<string> subjectAlternativeName,
+            string friendlyName,
+            DateTimeOffset notBefore,
+            DateTimeOffset expires,
+            StoreName storeName,
+            StoreLocation storeLocation)
+        {
+            using (var rsa = RSA.Create(2048))
+            {
+                var signingRequest = new CertificateRequest(
```

Comment:
@blowdart This is the cert creation part. It has basic constraints, EKUs, key usage and subject and san. Am I missing something?